### PR TITLE
Add Microsoft.CSharp source

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by Priya91** [comment]
:+1:  LGTM

**Comment by Priya91** [comment]
cc @joshfree

**Comment by AlekseyTs** [comment]
I believe I created a new project.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+                }
+            }
+            return false;
+        }
+
+        /////////////////////////////////////////////////////////////////////////////////
+
+        private static bool IsTransparentProxy(object obj)
+        {
+            return obj != null; // && RemotingServices.IsTransparentProxy(obj);
```
Comment (Maxwe11) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs]: I think here should be explanation why this commented

**Comment by AlekseyTs** [comment]
It looks like I copied the project after all, I'll change the guid and update the PR

**Comment by Maxwe11** [comment]
Why this project require `app.config`?

**Comment by AlekseyTs** [comment]
@Maxwe11 The app.config was created automatically. I'll try to remove it.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs (hunk 1 above)
Comment (mellinoe) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs]: Feel free to do this separately, but you can probably just remove the comment entirely. Seems to be related to Remoting, which isn't supported in .NET Core, and we've just been removing such things from other parts of the source.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+                }
+            }
+            return false;
+        }
+
+        /////////////////////////////////////////////////////////////////////////////////
+
+        private static bool IsTransparentProxy(object obj)
+        {
+            return obj != null;
```
Comment (eatdrinksleepcode) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs]: Removing the rest of the condition now means that every object passed to this method is considered a transparent proxy. Is that really the desired behavior? I would think that the method should just be changed to return false until all of its callers can be updated.

Comment (AlekseyTs) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs]: This is the current state of the source and I am not going to experiment with it.

**Comment by mellinoe** [comment]
This looks good to me, Aleksey, and the CI build succeeded. Feel free to merge it in.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs (hunk 2 above)
Comment (Maxwe11) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs]: Agree with @eatdrinksleepcode. Original code should be left but commented with explanation.

**Pull request merged**